internal/pkg/runtime/engines/singularity/container.go

```diff
@@ -322,27 +281,32 @@ func (c *container) setupSessionLayout(system *mount.System) error {
 		}
 	}
 
-	if c.engine.EngineConfig.GetWritableImage() && !writableTmpfs {
-		imgObject, err := c.loadImage(c.engine.EngineConfig.GetImage(), true)
-		if err != nil {
-			return err
-		}
+	imgObject, err := c.loadImage(c.engine.EngineConfig.GetImage(), true)
+	if err != nil {
+		return err
+	}
 
+	if c.engine.EngineConfig.GetWritableImage() && !writableTmpfs {
+		sylog.Debugf("Image is writable, not attempting to use overlay or underlay\n")
 		if imgObject.Type == image.SIF {
-			err = c.setupWritableSIFImage(imgObject, overlayEnabled)
+			err = c.setupSIFOverlay(imgObject, c.engine.EngineConfig.GetWritableImage())
 			if err == nil {
 				return c.setupOverlayLayout(system, sessionPath)
 			}
 			sylog.Warningf("%s", err)
-		} else {
-			sylog.Debugf("Image is writable, not attempting to use overlay or underlay\n")
 		}
-
 		return c.setupDefaultLayout(system, sessionPath)
 	}
 
 	if overlayEnabled {
 		sylog.Debugf("Attempting to use overlayfs (enable overlay = %v)\n", c.engine.EngineConfig.File.EnableOverlay)
+		if imgObject.Type == image.SIF {
+			err = c.setupSIFOverlay(imgObject, c.engine.EngineConfig.GetWritableImage())
+			if err == nil {
+				return c.setupOverlayLayout(system, sessionPath)
+			}
+			sylog.Warningf("%s", err)
```

Comment:
We need a more descriptive warning message here:

`sylog.Warningf("While attempting to set up SIFOverlay: %s", err)`